### PR TITLE
Add EditorConfig

### LINKED ISSUE #2540: Add .editorconfig for code formatting consistency
Happened to notice some code-formatting inconsistencies in regards to spaces and etc in older files. 

To make it easier for developers to stick to the [code formatting guidelines](http://docs.withknown.com/en/latest/developers/standards/#indentation), I would like to propose to introduce an ``.editorconfig`` file which would help set up the developer's IDE to stick to the project expected defaults.

An example `.editorconfig` file:
```
# EditorConfig is awesome: https://EditorConfig.org

# top-most EditorConfig file
root = true

# Unix-style newlines with a newline ending every file
[*]
indent_style = space
indent_size = 4
end_of_line = lf
insert_final_newline = true
````

See https://editorconfig.org/

cf. http://docs.withknown.com/en/latest/developers/standards/#indentation

### PR DESCRIPTION
**Pull request by thierrymarianne**
## Here's what I fixed or added:

ff4e2216024d77f1c349374f0c7a7f2f64b5f2e2 closes https://github.com/idno/known/issues/2540

## Checklist: (`[x]` to check/tick the boxes)

- [x] This pull request addresses a single issue
- [ ] If this code includes interface changes, I've included screenshots in this Pull Request thread
- [x] I've adhered to [Known's style guide](http://docs.withknown.com/en/latest/developers/standards/) ([these codesniffer rules](http://docs.withknown.com/en/latest/developers/testing/#code-style-testing) might help!)
- [x] My git branch is named in a descriptive way - i.e., yourname-summary-of-issue
- [x] I've tested my code in-browser
- [x] My code contains descriptive comments
- [ ] I've added tests where applicable, and...
- [x] I can run the [unit tests](http://docs.withknown.com/en/latest/developers/testing/#unit-testing) successfully.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "idno/known",
     "description": "A social publishing platform.",
-    "version": "1.3.1",
+    "version": "1.5.0",
```
Comment (lindner) [inline composer.json]: This should go in a separate commit.  Can you omit this?

Comment (thierrymarianne) [reply composer.json]: I've removed the commit responsible for this modification.

**Review by lindner — APPROVED** [review]
composer lock changes are a little gnarly, but should be fine.
